src/Compilers/Core/Portable/Operations/OperationExtensions.Clone.cs

```diff
@@ -0,0 +1,477 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    public static partial class OperationExtensions
+    {
+        private class Cloner : OperationVisitor<object, IOperation>
```

Comment:
How is it easier?  If this only happens with local references in for-loops, then why don't we just make three local-reference-expression operations in htat case and be done with it?